### PR TITLE
Rem 266

### PR DESCRIPTION
**Pull request by Alinadimmokdad**

### CODE REVIEW

File: src/components/Shared/GlobalIntegrationGrid.js [hunk 1]
```diff
@@ -135,6 +135,8 @@ const GlobalIntegrationGrid = ({ masterSource, masterId }) => {
     } catch (error) {}
   }
 
+  console.log(formik, 'aaaaa')
```
Comment (omarZrayka23) [inline src/components/Shared/GlobalIntegrationGrid.js:138]: remove console.log

File: src/pages/fi-tax-codes/forms/HistoryForm.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                  name: 'date'
+                },
+                {
+                  component: 'numberfield',
+                  label: labels.amount,
+                  name: 'amount',
+                  decimalScale: 2
+                }
+              ]}
+              height={`${expanded ? height - 280 : 380}px`}
```
Comment (omarZrayka23) [inline src/pages/fi-tax-codes/forms/HistoryForm.js:137]: do u need this?

File: src/pages/fi-tax-codes/index.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    paginationParameters,
+    refetch,
+    access
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: FinancialRepository.TaxCodes.qry,
+    datasetId: ResourceIds.TaxCodes
+  })
+
+  const invalidate = useInvalidate({
```
Comment (omarZrayka23) [inline src/pages/fi-tax-codes/index.js:44]: u can use invalidate in useResourceQuery

File: src/pages/fi-tax-codes/Window/TaxCodesWindow.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+const TaxCodesWindow = ({ height, recordId, labels, maxAccess, expanded }) => {
+  const [activeTab, setActiveTab] = useState(0)
+  const [editMode, setEditMode] = useState(recordId)
+
+  const [store, setStore] = useState({
+    recordId: recordId || null,
+    TaxHistoryViewList: []
+  })
+
+  const tabs = [{ label: labels.taxCodes }, { label: labels.history, disabled: !store.recordId }]
```
Comment (omarZrayka23) [inline src/pages/fi-tax-codes/Window/TaxCodesWindow.js:17]: name of second tab is hidden ?

File: src/pages/fi-tax-codes/Window/TaxCodesWindow.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        />
+      </CustomTabPanel>
+      <CustomTabPanel height={height} index={1} value={activeTab}>
+        <HistoryForm
+          labels={labels}
+          setEditMode={setEditMode}
+          setStore={setStore}
+          maxAccess={maxAccess}
+          store={store}
+          expanded={expanded}
```
Comment (omarZrayka23) [inline src/pages/fi-tax-codes/Window/TaxCodesWindow.js:40]: do u need this?

**Review by omarZrayka23 — CHANGES_REQUESTED** [review]

File: src/components/Shared/GlobalIntegrationGrid.js (hunk 1 above)
Comment (Alinadimmokdad) [reply src/components/Shared/GlobalIntegrationGrid.js]: done

File: src/pages/fi-tax-codes/forms/HistoryForm.js (hunk 2 above)
Comment (Alinadimmokdad) [reply src/pages/fi-tax-codes/forms/HistoryForm.js]: no

File: src/pages/fi-tax-codes/Window/TaxCodesWindow.js (hunk 5 above)
Comment (Alinadimmokdad) [reply src/pages/fi-tax-codes/Window/TaxCodesWindow.js]: no , removed

File: src/pages/fi-tax-codes/Window/TaxCodesWindow.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
+// ** Custom Imports
+import CustomTabPanel from 'src/components/Shared/CustomTabPanel'
+import { CustomTabs } from 'src/components/Shared/CustomTabs'
+import { useState } from 'react'
+import TaxCodesForm from '../forms/TaxCodesForm'
+import HistoryForm from '../forms/HistoryForm'
+
+const TaxCodesWindow = ({ height, recordId, labels, maxAccess }) => {
+  const [activeTab, setActiveTab] = useState(0)
+  const [editMode, setEditMode] = useState(recordId)
```
Comment (hadichahine) [inline src/pages/fi-tax-codes/Window/TaxCodesWindow.js:10]: edit mode

File: src/pages/fi-tax-codes/Window/TaxCodesWindow.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  const tabs = [{ label: labels.taxCodes }, { label: labels.history, disabled: !store.recordId }]
+
+  return (
+    <>
+      <CustomTabs tabs={tabs} activeTab={activeTab} setActiveTab={setActiveTab} />
+
+      <CustomTabPanel height={height} index={0} value={activeTab}>
+        <TaxCodesForm
+          labels={labels}
+          setEditMode={setEditMode}
```
Comment (hadichahine) [inline src/pages/fi-tax-codes/Window/TaxCodesWindow.js:26]: No need for setEditMode

File: src/pages/fi-tax-codes/forms/HistoryForm.js [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+import toast from 'react-hot-toast'
+import { useWindowDimensions } from 'src/lib/useWindowDimensions'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+
+const HistoryForm = ({ store, setStore, maxAccess, labels, editMode }) => {
+  const { recordId } = store
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { height } = useWindowDimensions()
```
Comment (hadichahine) [inline src/pages/fi-tax-codes/forms/HistoryForm.js:18]: Drop all uses of height

File: src/pages/fi-tax-codes/forms/HistoryForm.js [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+              id: index + 1,
+              date: formatDateFromApi(item.date),
+              amount: item.amount
+            }))
+            formik.setValues({ TaxHistoryView: items })
+            setStore(prevStore => ({
+              ...prevStore,
+              TaxHistoryView: items
+            }))
+          } else {
```
Comment (hadichahine) [inline src/pages/fi-tax-codes/forms/HistoryForm.js:102]: We already fixed this in the previous commit. Do the same here.

File: src/pages/fi-tax-codes/forms/TaxCodesForm.js [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { FinancialRepository } from 'src/repositories/FinancialRepository'
+import { MasterSource } from 'src/resources/MasterSource'
+
+export default function TaxCodesForm({ labels, maxAccess, setStore, store }) {
+  const { recordId } = store
+  const [editMode, setEditMode] = useState(!!recordId)
```
Comment (hadichahine) [inline src/pages/fi-tax-codes/forms/TaxCodesForm.js:18]: Check if we can get this from props.

File: src/pages/fi-tax-codes/Window/TaxCodesWindow.js (hunk 6 above)
Comment (hadichahine) [inline src/pages/fi-tax-codes/Window/TaxCodesWindow.js:10]: edit mode

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/pages/fi-tax-codes/Window/TaxCodesWindow.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
+// ** Custom Imports
+import CustomTabPanel from 'src/components/Shared/CustomTabPanel'
+import { CustomTabs } from 'src/components/Shared/CustomTabs'
+import { useState } from 'react'
+import TaxCodesForm from '../forms/TaxCodesForm'
+import HistoryForm from '../forms/HistoryForm'
+
+const TaxCodesWindow = ({ height, recordId, labels, maxAccess }) => {
```
Comment (hadichahine) [inline src/pages/fi-tax-codes/Window/TaxCodesWindow.js:8]: Why do we need height here?

**Review by hadichahine — CHANGES_REQUESTED** [review]

**Review by hadichahine — APPROVED** [review]

**Pull request merged**